Source/WebKit/_WebKit_SwiftUI/ViewModifierContexts.swift

```diff
@@ -30,9 +30,45 @@ struct ContextMenuContext {
 #endif
 }
 
-struct OnScrollGeometryChangeContext {
-    let transform: (ScrollGeometry) -> AnyEquatable
-    let action: (AnyEquatable, AnyEquatable) -> Void
+class OnScrollGeometryChangeContext {
+    private struct Change {
+        var transform: (ScrollGeometry) -> AnyHashable
+        var action: (AnyHashable, AnyHashable) -> Void
+    }
+
+    private var changes: [Namespace.ID : Change] = [:]
+
+    func register<T>(
+        changeID: Namespace.ID,
+        transform: @escaping (ScrollGeometry) -> T,
+        action: @escaping (T, T) -> Void
+    ) where T: Hashable {
+        let erasedTransform = { (geometry: ScrollGeometry) in
+            AnyHashable(transform(geometry))
+        }
+
+        let erasedAction = { (old: AnyHashable, new: AnyHashable) in
+            action(old.base as! T, new.base as! T)
+        }
+
+        if changes[changeID] == nil {
```

Comment:
I think that without some sort of `= nil` call site, you’re going to be leaking the change observers here.

Reply:
Thanks for catching this! I added a FIXME and will address in a subsequent commit.